dexlib2/src/main/java/org/jf/dexlib2/writer/DexWriter.java

```diff
@@ -503,8 +513,11 @@ private void writeClasses(@Nonnull DexDataStore dataStore, @Nonnull DexDataWrite
             return;
         }
 
+        offsetWriter.align();
         hiddenApiRestrictionsOffset = offsetWriter.getPosition();
 
+        Collections.sort(classEntries, DexWriter.<ClassKey>comparableValueComparator());
+
```

Comment:
what are the consequences of leaving `classEntries` with an order that depends on `shouldWriteHiddenApiRestrictions()`? could sorting a copy be preferable?

Reply:
yeah that would be fine, I believe. it's just for the hiddenapi section.